quinn-h3/examples/h3.rs

```diff
@@ -291,31 +291,50 @@ fn client(
 
             conn.request(request)
                 .trailers(trailer)
-                .send()
-                .map_err(|e| format_err!("send request: {}", e))
-                .and_then(|response| {
-                    println!("received response: {:?}", response);
-                    let (_, body) = response.into_parts();
-                    let buf = Vec::with_capacity(1024 * 10); // 10K
-                    tokio_io::io::read_to_end(body.into_reader(), buf)
+                .stream()
+                .map_err(|e| format_err!("send request failed: {}", e))
+                .and_then(|(send_body, response)| {
+                    let response_fut = response
                         .map_err(|e| format_err!("receive response failed: {}", e))
-                        .and_then(|(reader, data)| {
-                            println!("received body len = {}", data.len());
-                            if let Some(decode_trailers) = reader.trailers() {
-                                return Either::A(
-                                    decode_trailers
-                                        .map_err(|e| format_err!("decode trailers failed: {}", e))
-                                        .and_then(|trailers| {
-                                            println!(
-                                                "received trailers: {:?}",
-                                                trailers.into_fields()
-                                            );
-                                            futures::future::ok(())
-                                        }),
-                                );
-                            }
-                            Either::B(futures::future::ok(()))
-                        })
+                        .and_then(|response| {
+                            println!("received response: {:?}", response);
+
+                            let (_, body) = response.into_parts();
+
+                            let buf = Vec::with_capacity(1024 * 10); // 10K
+                            tokio_io::io::read_to_end(body.into_reader(), buf)
```

Comment:
This looks like things have gotten a good deal more complex. Should we add some abstractions to make the common cases easier to deal with?

Reply:
That's right. I'm not sure of what I can do to get a simpler API. 
Still, this example is the most complex thing you can acheive as a user right now: sending request body and recieving response body in parallel.

A simpler API is still availabe through `ReqBuilder::send()`. 

I'll write a few more examples to show that way of doing things. This might help finding some more ergonomic solutions.

Reply:
I just pushed simpler examples, still more complex than when receiving body was in the same future as receiving response header.